corehq/apps/sms/views.py

```diff
@@ -1039,11 +1050,15 @@ def backend_class_name(self):
     def ignored_fields(self):
         return [
             'give_other_domains_access',
+            'phone_numbers',
         ]
 
     @property
     @memoized
     def backend_class(self):
+        # For now, only allow superusers to create/edit non-Telerivet backends
+        if not self.is_superuser and self.backend_class_name != "TelerivetBackend":
```

Comment:
@millerdev Never mind, I see where the comment was misleading - just updated it to be clearer.